plugins/module_utils/common.py

```diff
@@ -708,6 +708,11 @@ def perform_action(self, resource, definition):
                         if self.warnings:
                             msg += "\n" + "\n    ".join(self.warnings)
                         self.fail_json(msg=msg, error=exc.status, status=exc.status, reason=exc.reason)
+                    except Exception as exc:
```

Comment:
I would prefer we just catch the `ValueError`. This way if another valide exception is raised, it won't be hidden.